lib/async/io/ssl_endpoint.rb

```diff
@@ -89,8 +89,8 @@ def connect(&block)
 			def each
 				return to_enum unless block_given?
 				
-				@endpoint.each do |endpoint|
-					yield self.class.new(endpoint, @options)
+				@endpoint.each do |ep|
```

Comment:
One more here, perhaps use `singular_endpoint` to be consistent with the previous change if you think that makes sense.

Reply:
If you can nail these last changes, I can merge!

Reply:
I used the `singular_endpoint` here, too.